src/SonarScanner.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```
@@ -348,6 +390,72 @@
     <Message Importance="high" Text="Sonar: ($(MSBuildProjectFile)) Project processed successfully" />
   </Target>
 
+  <!-- Set the Razor specific error log properties and copy the results of the core compilation to a temporary folder.
+       Rename the output directory to "$(ProjectSpecificOutDir).tmp" to prevent RazorCoreCompilation from overriding its content.
+       It will be renamed back in SonarFinishRazorCodeAnalysis target.
+  -->
+  <Target Name="SonarPrepareRazorCodeAnalysis" BeforeTargets="RazorCoreCompile" AfterTargets="CoreCompile" Condition="'@(RazorCompile)'!=''" >
+    <PropertyGroup Condition=" $(SonarErrorLog) != '' ">
+      <!-- https://github.com/SonarSource/sonar-scanner-msbuild/issues/1015 Respect user-provided value. When set, users should provide unique path for each build (project and target framework). -->
+      <RazorCompilationErrorLogName Condition=" $(RazorCompilationErrorLog) == '' ">Issues$(RazorTargetNameSuffix).json</RazorCompilationErrorLogName>
+      <RazorCompilationErrorLog Condition=" $(RazorCompilationErrorLog) == '' ">$(ProjectSpecificOutDir)\Issues$(RazorTargetNameSuffix).json</RazorCompilationErrorLog>
```

Should reuse the computed value.
And should probably use also the same condition, as that is our state now to detect if we should do it or not.
```suggestion
      <RazorCompilationErrorLog Condition=" $(RazorCompilationErrorLogName) != '' ">$(ProjectSpecificOutDir)\$(RazorCompilationErrorLogName)</RazorCompilationErrorLog>
```